app/models/api_token.rb

```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal: true
 
 class ApiToken < ApplicationRecord
+  # This is meant to be an abstract class
+  # Since it is a base class for a STI, we can't actually make it abstract (not backed by a table)
```

Comment:
I think `validates_presence_of :type` here would prevent storing an instance of ApiToken. See https://stackoverflow.com/questions/2850418/rails-sti-prevent-base-class-from-instantiation/4834082#4834082

Happy to leave this as is if you want though

Reply:
I tried that actually.

The PR that introduced `ApiToken` and STI on it had a problem: how do we migrate existing lead provider tokens into new api tokens? And I chose the easy way out - I added a default value to `type`, namely `LeadProviderApiToken`. So that check won't work 😢 

I guess I could write a migration to undo that, but that feels like significantly more effort than it's worth.

Reply:
Ah... fair enough 🙂